src/components/scene/device-motion-permission-ui.js

```diff
@@ -0,0 +1,159 @@
+var registerComponent = require('../../core/component').registerComponent;
+var utils = require('../../utils/');
+var bind = utils.bind;
+
+var constants = require('../../constants/');
+
+var DEVICE_PERMISSION_FULL_CLASS = 'a-device-motion-permission-full';
+var DEVICE_PERMISSION_FULL_CENTER_CLASS = 'a-device-motion-permission-full-center';
+var DEVICE_PERMISSION_CONTINUE_CLASS = 'a-device-motion-permission-continue';
+var DEVICE_PERMISSION_CANCEL_CLASS = 'a-device-motion-permission-cancel';
+var BUILT_WITH_AFRAME_CLASS = 'a-built-with-aframe';
+
+/**
+ * UI for enabling device motion permission
+ */
+module.exports.Component = registerComponent('device-motion-permission-ui', {
+  schema: {
+    enabled: { default: true },
+    request: {default: {orientationChange:null, deviceOrientation:null}},
+    deviceMotionEl: { default: '' }
+  },
+
+  init: function () {
+    this.deviceMotionEl = null;
+    this.onDeviceMotionClick = bind(this.onDeviceMotionClick, this);
+    this.onOrientationChangeClick = bind(this.onOrientationChangeClick, this);
+    this.grantedDeviceMotion = bind(this.grantedDeviceMotion, this);
+    if (typeof window.orientation !== 'undefined') {
+      try {
+        /*eslint-disable */
+        if (
+          DeviceOrientationEvent &&
+          typeof DeviceOrientationEvent.requestPermission === 'function'
+        ) {
+          DeviceOrientationEvent.requestPermission()
+            .then(response => {
+              if (response === 'granted') {
+                this.grantedDeviceMotion();
+              }
+            })
+            /* eslint-enable */
+            .catch(err => {
+              console.log(err);
+              this.deviceMotionEl = createDeviceMotionPermissionWindow(
+                this.onDeviceMotionClick,
+                this
+              );
+              this.el.appendChild(this.deviceMotionEl);
+            });
+        } else {
+          this.grantedDeviceMotion();
+        }
+      } catch (oops) {
+        this.grantedDeviceMotion();
+      }
+    } else {
+      this.remove();
+    }
+  },
+
+  remove: function () {
+    if (this.deviceMotionEl) {
+      this.el.removeChild(this.deviceMotionEl);
+    }
+  },
+
+  /**
+   * Enable device motion permission when clicked.
+   */
+  onDeviceMotionClick: function () {
+    try {
+      /*eslint-disable */
```

Comment:
Don't use `eslint-disable`. Linter issues should be addressed

Reply:
This has been updated to omit eslint-disable from the code.